src/diffusers/loaders/single_file.py

```diff
@@ -245,6 +252,7 @@ def from_single_file(cls, pretrained_model_link_or_path, **kwargs):
         load_safety_checker = (kwargs.pop("load_safety_checker", False)) or (
             passed_class_obj.get("safety_checker", None) is not None
         )
+        model_type = "Playground" if "edm_mean" in checkpoint and "edm_std" in checkpoint else model_type
```

Comment:
Can we move inferring the model type into this function? Can update it to accept checkpoint as well. It would keep model type inference logic isolated to that function.  
https://github.com/huggingface/diffusers/blob/6d3e82c9cd00483104a6f405d5e5893c2bed5c20/src/diffusers/loaders/single_file_utils.py#L310

Reply:
> Can we move inferring the model type into this function? Can update it to accept checkpoint as well. It would keep model type inference logic isolated to that function.
> 
> https://github.com/huggingface/diffusers/blob/6d3e82c9cd00483104a6f405d5e5893c2bed5c20/src/diffusers/loaders/single_file_utils.py#L310

s it possible to push this fix to A1111? really need it, tks!